Sprint 10 ver. 1

DmitryBrosalin opened the pull request:
Первая версия задания по спринту 10. Фильморейт.

DmitryBrosalin committed:
Sprint 10 ver. 1
Первая версия задания по спринту 10. Фильморейт.

DmitryBrosalin committed:
Sprint 10 ver. 1
Первая версия задания по спринту 10. Фильморейт.

DmitryBrosalin committed:
Sprint 10 ver. 1
Первая версия задания по спринту 10. Фильморейт.

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+    }
+
+    @PutMapping
+    public Film upfateFilm(@Valid @RequestBody Film film) {
```
Опечатка: `updateFilm`

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+            throw new NotFoundException("Film id not found");
+        }
+        if (film.getReleaseDate().isBefore(LocalDate.of(1895, 12, 28))) {
+            throw new ValidationException("Release Date must not be before 28/12/1895");
```
Если будет желание, для таких проверок можно создать кастомные аннотации https://www.baeldung.com/spring-mvc-custom-validator

slavrikk commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,65 @@
+            throw new ValidationException("Release Date must not be before 28/12/1895");
+        }
+        if (film.getDescription().length() > 200) {
+            throw new ValidationException("Description length must be 200 or less");
```
Аналогичный комментарий

DmitryBrosalin committed:
Sprint 10 ver. 2
- Исправил опечатку updateFilm.
- Добавил кастомные аннотации для валидации полей.

DmitryBrosalin committed:
Sprint 10 ver. 2
- Исправил опечатку updateFilm.
- Добавил кастомные аннотации для валидации полей.
